app/advancedfilters/annotationExtensionFilter.js

```diff
@@ -35,6 +35,12 @@ app.controller('annotationExtensionFilterController', function($scope, $rootScop
     return !$scope.relationship && !$scope.db;
   }
 
+  $scope.updateInputBox = function(textBoxId, source){
```

Comment:
This is unnecessary...in angularJS the `model` is updated automatically. You shouldn't modify these files but look into `autocomplete.js` instead. 
Currently `scope.searchTerm` only updates `scope.model` when an item is selected, it should be updated regardless.